lifetiles-graph/src/main/java/nl/tudelft/lifetiles/annotation/controller/BookmarkController.java

```diff
@@ -0,0 +1,181 @@
+                     */
+                    public void updateItem(final AbstractBookmark item,
+                            final boolean empty) {
+                        super.updateItem(item, empty);
+                        if (item != null) {
+                            setText(item.toCellString());
+
+                            setOnMouseClicked(event -> {
+                                if (event.getClickCount() == 2) {
+                                    shout(Message.GOTO, "" ,
+                                            Long.valueOf(item.getUnifiedPosition()));
+                                }
+                            });
+                        }
+                    }
+                };
+                return cell;
+            }
+        });
+        initListeners();
+    }
+
+
+    /**
+     * Initialize the shout listeners.
+     */
+    // java doesn't let us typecheck generics so the casts are unchecked
+    // checkstyle doesn't like assert without parentheses with generics
+    @SuppressWarnings({"checkstyle:genericwhitespace", "unchecked"})
+    private void initListeners() {
+        listen(Message.BOOKMARKS, (sender, subject, args) -> {
+            show();
+        });
+
+        listen(Message.LOADED, (sender, subject, args) -> {
+
+            switch (subject) {
+            case "known mutations":
+                assert args[0] instanceof List<?>;
+                List<KnownMutation> knownMutations = (List<KnownMutation>) args[0];
+                listItems.addAll(knownMutations);
+                return;
+            case "annotations":
+                assert args[0] instanceof List<?>;
+                List<GeneAnnotation> genes = (List<GeneAnnotation>) args[0];
+                listItems.addAll(genes);
+                return;
+            default:
+                return;
+            }
+        });
+    }
+    /**
+     * closes the bookmark sidebar.
+     */
+    @FXML
+    // PMD/findbugs do not work well with javafx. The method IS used.
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD")
+    private void closeAction() {
+        hide();
+    }
+
+    /**
+     * updates the filtering predicate.
+     * uses a simple contains check on the text of the listItem.
+     */
+    @FXML
+    // PMD/findbugs do not work well with javafx. The method IS used.
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD")
+    private void searchAction() {
+        filteredItems.setPredicate(annotation -> {
+            if (searchBox.textProperty().get().isEmpty()) {
+                return true;
+            }
+           return annotation.toCellString().contains(searchBox.getCharacters());
+        });
+    }
+
+    /**
+     * makes the sidebar invisible.
+     */
+    private void hide() {
+        wrapper.visibleProperty().set(false);
```

Comment:
`wrapper.visibleProperty().set(...)` is duplicaate code, it should be moved to a method that takes a boolean as argument